diff --git a/flash_layout.h b/flash_layout.h
index 0974e06..134e79a 100644
--- a/flash_layout.h
+++ b/flash_layout.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018-2022 Arm Limited. All rights reserved.
+ * Copyright (c) 2018-2021 Arm Limited. All rights reserved.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,30 +16,27 @@
 
 #ifndef __FLASH_LAYOUT_H__
 #define __FLASH_LAYOUT_H__
-/* new field for OSPI */
 
-#define OSPI_FLASH_TOTAL_SIZE           (0x4000000)  /* 64 MB same as MX25LM51245G_FLASH_SIZE */
-#define OSPI_FLASH_BASE_ADDRESS         (0x70000000) /* same as OCTOSPI1_BASE  */
-#define EXTERNAL_FLASH
 /* This header file is included from linker scatter file as well, where only a
  * limited C constructs are allowed. Therefore it is not possible to include
  * here the platform_retarget.h to access flash related defines. To resolve this
  * some of the values are redefined here with different names, these are marked
  * with comment.
  */
+
 /* Flash layout for b_u585i_iot02a  with BL2 (multiple image boot):
  *
- * 0x0000_0000 SCRATCH (64KB)
- * 0x0001_0000 BL2 - counters(16 KB)
- * 0x0001_4000 BL2 - MCUBoot (84 KB)
- * 0x0002_7000 OTP Write Protect (4KB)
- * 0x0002_8000 NV counters area (16 KB)
- * 0x0002_c000 Secure Storage Area (16 KB)
- * 0x0003_0000 Internal Trusted Storage Area (16 KB)
- * 0x0003_8000 Secure image     primary slot (384 KB)
- * 0x0009_8000 Non-secure image primary slot (512 KB)
- * 0x0011_8000 Secure image     secondary slot (384 KB)
- * 0x0017_8000 Non-secure image secondary slot (512 KB)
+ * 0x0000_0000 SCRATCH                          (64KB)      0x10000
+ * 0x0001_0000 BL2 - counters                   (16 KB)     0x04000
+ * 0x0001_4000 BL2 - MCUBoot                    (124 KB)    0x1F000
+ * 0x0003_3000 OTP Write Protect                (4KB)       0x01000
+ * 0x0002_E000 NV counters area                 (16 KB)     0x04000
+ * 0x0003_2000 Secure Storage Area              (16 KB)     0x04000
+ * 0x0003_6000 Internal Trusted Storage Area    (16 KB)     0x04000
+ * 0x0003_A000 Secure image     primary slot    (256 KB)    0x40000
+ * 0x0007_A000 Non-secure image primary slot    (640 KB)    0xA0000
+ * 0x0011_A000 Secure image     secondary slot  (256 KB)    0x40000
+ * 0x0015_A000 Non-secure image secondary slot  (640 KB)    0xA0000
  *
  * Bl2 binary is written at 0x1_2000:
  * it contains bl2_counter init value, OTP write protect, NV counters area init.
@@ -53,7 +50,6 @@
 
 /* Flash device ID */
 
-
 /* Offset and size definitions of the flash partitions that are handled by the
  * bootloader. The image swapping is done between IMAGE_0 and IMAGE_1, SCRATCH
  * is used as a temporary storage during image swapping.
@@ -70,9 +66,11 @@
 
 /* area for bl2 anti roll back counter */
 #define FLASH_BL2_NVCNT_AREA_OFFSET     (FLASH_AREA_SCRATCH_SIZE)
-#define FLASH_BL2_NVCNT_AREA_SIZE       (FLASH_AREA_IMAGE_SECTOR_SIZE+FLASH_AREA_IMAGE_SECTOR_SIZE)
+#define FLASH_BL2_NVCNT_AREA_SIZE       (0x4000)
+
 /* Area for downloading bl2 image */
 #define FLASH_AREA_BL2_BIN_OFFSET       (FLASH_BL2_NVCNT_AREA_OFFSET+FLASH_AREA_IMAGE_SECTOR_SIZE)
+
 /* personal Area Not used */
 #define FLASH_AREA_PERSO_OFFSET         (FLASH_BL2_NVCNT_AREA_OFFSET+FLASH_BL2_NVCNT_AREA_SIZE)
 #define FLASH_AREA_PERSO_SIZE           (0x0)
@@ -83,23 +81,27 @@
 
 /* area for BL2 code protected by hdp */
 #define FLASH_AREA_BL2_OFFSET           (FLASH_AREA_PERSO_OFFSET+FLASH_AREA_PERSO_SIZE)
-#define FLASH_AREA_BL2_SIZE             (0x16000)
+#define FLASH_AREA_BL2_SIZE             (0x18000)
+
 /* HDP area end at this address */
 #define FLASH_BL2_HDP_END               (FLASH_AREA_BL2_OFFSET+FLASH_AREA_BL2_SIZE-1)
+
 /* area for BL2 code not protected by hdp */
 #define FLASH_AREA_BL2_NOHDP_OFFSET     (FLASH_AREA_BL2_OFFSET+FLASH_AREA_BL2_SIZE)
 #define FLASH_AREA_BL2_NOHDP_CODE_SIZE  (0x1000)
 #define FLASH_AREA_OTP_OFFSET           (FLASH_AREA_BL2_NOHDP_OFFSET+FLASH_AREA_BL2_NOHDP_CODE_SIZE)
 #define FLASH_AREA_OTP_SIZE             (0x1000)
 #define FLASH_AREA_BL2_NOHDP_SIZE       (FLASH_AREA_OTP_SIZE+FLASH_AREA_BL2_NOHDP_CODE_SIZE)
+
 /* control area for BL2 code protected by hdp */
 #if (FLASH_AREA_BL2_NOHDP_OFFSET % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0
 #error "HDP area must be aligned on FLASH_AREA_IMAGE_SECTOR_SIZE"
 #endif /* (FLASH_AREA_BL2_NOHDP_OFFSET % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0 */
 
 /* Non Volatile Counters definitions */
-#define FLASH_NV_COUNTERS_AREA_SIZE      (FLASH_AREA_IMAGE_SECTOR_SIZE+FLASH_AREA_IMAGE_SECTOR_SIZE)
 #define FLASH_NV_COUNTERS_AREA_OFFSET    (FLASH_AREA_BL2_NOHDP_OFFSET+FLASH_AREA_BL2_NOHDP_SIZE)
+#define FLASH_NV_COUNTERS_AREA_SIZE      (0x4000)
+
 /* Control Non Volatile Counters definitions */
 #if (FLASH_NV_COUNTER_AREA_SIZE % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0
 #error "FLASH_NV_COUNTER_AREA_SIZE not aligned on FLASH_AREA_IMAGE_SECTOR_SIZE"
@@ -123,16 +125,15 @@
 #error "FLASH_ITS_AREA_OFFSET not aligned on FLASH_AREA_IMAGE_SECTOR_SIZE"
 #endif /*  (FLASH_ITS_AREA_OFFSET % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0 */
 
-#define FLASH_S_PARTITION_SIZE          (0x60000) /* 384 KB for S partition */
-#define FLASH_NS_PARTITION_SIZE         (0x80000) /* 512 KB for NS partition */
+#define FLASH_S_PARTITION_SIZE          (0x40000) /* 256 KB for S partition */
+#define FLASH_NS_PARTITION_SIZE         (0xA0000) /* 640 KB for NS partition */
 
 #define FLASH_PARTITION_SIZE            (FLASH_S_PARTITION_SIZE+FLASH_NS_PARTITION_SIZE)
 
-#if (FLASH_S_PARTITION_SIZE > FLASH_NS_PARTITION_SIZE)
-#define FLASH_MAX_PARTITION_SIZE FLASH_S_PARTITION_SIZE
-#else
-#define FLASH_MAX_PARTITION_SIZE FLASH_NS_PARTITION_SIZE
-#endif
+#define FLASH_MAX_PARTITION_SIZE        ((FLASH_S_PARTITION_SIZE >   \
+                                          FLASH_NS_PARTITION_SIZE) ? \
+                                         FLASH_S_PARTITION_SIZE : \
+                                         FLASH_NS_PARTITION_SIZE)
 /* Secure image primary slot */
 #define FLASH_AREA_0_ID                 (1)
 #define FLASH_AREA_0_DEVICE_ID          (FLASH_DEVICE_ID-FLASH_DEVICE_ID)
@@ -156,11 +157,7 @@
 /* Secure image secondary slot */
 #define FLASH_AREA_2_ID                 (FLASH_AREA_1_ID+1)
 #define FLASH_AREA_2_DEVICE_ID          (FLASH_AREA_1_DEVICE_ID)
-#if defined(EXTERNAL_FLASH)
-#define FLASH_AREA_2_OFFSET             (0x000000000)
-#else
 #define FLASH_AREA_2_OFFSET             (FLASH_AREA_1_OFFSET+FLASH_AREA_1_SIZE)
-#endif /* EXTERNAL FLASH */
 /* Control  Secure image secondary slot */
 #if (FLASH_AREA_2_OFFSET  % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0
 #error "FLASH_AREA_2_OFFSET  not aligned on FLASH_AREA_IMAGE_SECTOR_SIZE"
@@ -170,12 +167,8 @@
 /* Non-secure image secondary slot */
 #define FLASH_AREA_3_ID                 (FLASH_AREA_2_ID+1)
 #define FLASH_AREA_3_DEVICE_ID          (FLASH_AREA_2_DEVICE_ID)
-#if defined(EXTERNAL_FLASH)
-/* Add 0x8000 to fix tools issue on external flash */
-#define FLASH_AREA_3_OFFSET             (FLASH_AREA_2_OFFSET + FLASH_AREA_2_SIZE + 0x8000)
-#else
 #define FLASH_AREA_3_OFFSET             (FLASH_AREA_2_OFFSET+FLASH_AREA_2_SIZE)
-#endif /* EXTERNAL FLASH */
+
 #if (FLASH_AREA_3_OFFSET  % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0
 #error "FLASH_AREA_3_OFFSET  not aligned on FLASH_AREA_IMAGE_SECTOR_SIZE"
 #endif /*  (FLASH_AREA_3_OFFSET  % FLASH_AREA_IMAGE_SECTOR_SIZE) != 0 */
@@ -185,18 +178,6 @@
 #define FLASH_AREA_SCRATCH_ID           (FLASH_AREA_3_ID+1)
 #define FLASH_AREA_SCRATCH_DEVICE_ID    (FLASH_AREA_3_DEVICE_ID)
 
-
-#if defined(EXTERNAL_FLASH)
-/* Config for Area Using External flash driver */
-#define OSPI_FLASH_DEV_ID          (FLASH_DEVICE_ID+1)
-#define FLASH_DEVICE_ID_2          (OSPI_FLASH_DEV_ID)
-#define FLASH_DEVICE_ID_3          (OSPI_FLASH_DEV_ID)
-#define OSPI_FLASH_DEV_NAME   TFM_Driver_OSPI_FLASH0
-#define FLASH_DEV_NAME_2 OSPI_FLASH_DEV_NAME
-#define FLASH_DEV_NAME_3 OSPI_FLASH_DEV_NAME
-
-#define FLASH_DRIVER_LIST {&TFM_Driver_OSPI_FLASH0, &TFM_Driver_FLASH0}
-#endif /* defined(EXTERNAL_FLASH) */
 /*
  * The maximum number of status entries supported by the bootloader.
  */
